interfaces/prompting/errors/errors.go

```diff
@@ -0,0 +1,242 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// The errors package defines common error types which are used across the
+// prompting subsystems, along with constructors for specific errors based on
+// those broader types.
+package errors
+
+import (
+	"errors"
+	"fmt"
+	"time"
+)
+
+var (
+	// NotFound errors when a prompt or rule is not found
+	ErrPromptNotFound = errors.New("cannot find prompt with the given ID for the given user")
+	ErrRuleNotFound   = errors.New("cannot find rule with the given ID")
+	ErrRuleNotAllowed = errors.New("user not allowed to request the rule with the given ID")
+
+	// Unsupported value errors, which are defined as functions below
+	ErrInvalidOutcome       = invalidOutcomeError
+	ErrInvalidLifespan      = invalidLifespanError
+	ErrRuleLifespanSingle   = ruleLifespanSingleError
+	ErrInvalidInterface     = invalidInterfaceError
+	ErrInvalidPermissions   = invalidPermissionsError
+	ErrPermissionsListEmpty = permissionsListEmptyError
+	// Parse errors, which are defined as functions below
+	ErrInvalidDuration    = invalidDurationError
+	ErrInvalidExpiration  = invalidExpirationError
+	ErrInvalidPathPattern = invalidPathPatternError
```

Comment:
let's keep this simple, you don't really need these variables, since they are not actual errors. The accepted practice is to have `ErrFooBar` be actual error values rather than functions, and then perhaps `NewFooBarError` if you need a helper that would provide additional context for the error value. Moreover, it'd be quite confusing if someone was to use these values with errors.Is() and did not notice the note that those are actually functions.

What I would suggest is to simply drop these...

```suggestion
```

Reply:
Very good point about the confusion using `errors.Is()`, thanks!